psydac/api/equation.py

```diff
@@ -93,7 +93,7 @@ def l2_boundary_projection(equation):
     test_dict = dict(zip(u, v))
 
     # Compute product of (u, v) using dot product for vector quantities
-    product  = lambda f, g: (f * g if isinstance(g, ScalarFunction) else dot(f, g))
+    product  = lambda f, g: (f * g if g.atoms(ScalarFunction) else dot(f, g))
```

Comment:
Please explain

Reply:
The previous implementation works only with homogeneous Neumann boundary conditions, so I changed it to add the case of non-homogeneous Neumann boundary conditions

Reply:
Was this happening for the biharmonic equation?

Reply:
I see thanks